internal/pkg/daemon/bpfrecorder/bpfrecorder.go

```diff
@@ -312,10 +312,65 @@ func (b *BpfRecorder) SyscallsForProfile(
 		return nil, errors.New("bpf recorder not running")
 	}
 	if b.Seccomp == nil {
-		return nil, errors.New("not recording seccomp profiles")
+		return nil, errors.New("not seccomp profiles recording running")
 	}
 	b.logger.Info("Getting syscalls for profile " + r.Name)
 
+	mntns, err := b.getMntnsForProfileWithRetry(r.Name)
+	if err != nil {
+		return nil, err
+	}
+	b.loadUnloadMutex.RLock()
+	syscalls, err := b.Seccomp.PopSyscalls(b, mntns)
+	b.loadUnloadMutex.RUnlock()
+	if err != nil {
+		b.logger.Error(err, "Failed to get syscalls for mntns", "mntns", mntns)
+		return nil, err
+	}
+
+	return &api.SyscallsResponse{
+		Syscalls: syscalls,
+		GoArch:   runtime.GOARCH,
+	}, nil
+}
+
+func (b *BpfRecorder) ApparmorForProfile(
+	_ context.Context, r *api.ProfileRequest,
+) (*api.ApparmorResponse, error) {
+
+	if b.startRequests == 0 {
+		return nil, errors.New("bpf recorder not running")
+	}
+	if b.AppArmor == nil {
+		return nil, errors.New("no apparmor profiles recording running")
+	}
+	b.logger.Info("Getting apparmor profile for profile " + r.Name)
+
+	mntns, err := b.getMntnsForProfileWithRetry(r.Name)
+	if err != nil {
+		return nil, err
+	}
+	b.loadUnloadMutex.RLock()
+	apparmor := b.AppArmor.GetAppArmorProcessed(mntns)
```

Comment:
For Seccomp we have `PopSyscalls` (which also removes the recorded events for the specific mntns). Should we do `PopAppArmorProcessed` as well?

Reply:
I think this is not necessary for apparmor because the events are streamed from eBPF into the user space while for seccomp the intercepted syscalls are stored directly into the map by the eBPF program. This map is shared and needs to be cleaned when the syscalls are read form the user space. This is the reason for `PopSyscalls`.